cmd/generate/config/rules/huggingface.go

```diff
@@ -0,0 +1,115 @@
+package rules
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/zricethezav/gitleaks/v8/cmd/generate/secrets"
+	"github.com/zricethezav/gitleaks/v8/config"
+)
+
+// Reference: https://huggingface.co/docs/hub/security-tokens
+//
+// Old tokens have the prefix `api_`, however, I am not sure it's worth detecting them as that would be high noise.
+// https://huggingface.co/docs/api-inference/quicktour
+func HuggingFaceAccessToken() *config.Rule {
+	// define rule
+	r := config.Rule{
+		RuleID:      "huggingface-access-token",
+		Description: "Hugging Face Access token",
+		Regex:       regexp.MustCompile(`(?:^|[\\'"` + "`" + ` >=:])(hf_[a-zA-Z]{34})(?:$|[\\'"` + "`" + ` <])`),
```

Comment:
@rgmz slightly concerned here since it's using regexp directly instead of one of the predefined generated functions. Any reason this can't be something like:
```
Regex: generateSemiGenericRegex([]string{"hf_", "hugging"}, "hf_" + alphaNumeric("34")),
```

Reply:
Both [`generateSemiGenericRegex`](https://github.com/gitleaks/gitleaks/blob/06db3b91d7495a39caf0d922aef5fee0118e7538/cmd/generate/config/rules/rule.go#L33) and [`generateUniqueTokenRegex`](https://github.com/gitleaks/gitleaks/blob/06db3b91d7495a39caf0d922aef5fee0118e7538/cmd/generate/config/rules/rule.go#L46) and the pattern case-insensitive, which increased the number of false positives I encountered during testing. I also found several instances of the token that wasn't detected with the existing predefined regexes (e.g., `"    hf_token = \"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\"\n",`).

That being said, I recognize that it's silly to hand-roll all the regexep boundaries on a per-rule basis. I was actually in midst of writing an issue describing how we could enhance the validation process by generation realistic scenarios that a secret could be found in.  Currently the quality of test data for true & false positives varies based on rule, and would greatly benefit from a set of consistent test data rather than people adding new symbols in an ad-hoc untestable way that.